details.js

```diff
@@ -0,0 +1,38 @@
+document.addEventListener("DOMContentLoaded", () => {
+    
+    const urlParams = new URLSearchParams(window.location.search);
+    const name = urlParams.get("name");
+    const flag = urlParams.get("flag");
+    const population = urlParams.get("population");
```

Comment:
You should extract this into service and use his methods from there, for example : 

`paramsService.getValue()`